### ISSUE TITLE
Need a new semantic property to force non-focus on widgets that have a value/label

### ISSUE THREAD

**Issue by ash2moon** [flutter/flutter#165857]
Affected platform: Android

## Context
In the AccessiblityService, the [`isFocusable`](https://github.com/flutter/flutter/blob/ecbb115ae3a8cba2977ecab9f52086df860cfb1a/engine/src/flutter/shell/platform/android/io/flutter/view/AccessibilityBridge.java#L2715) method allows a widget to be permanently focusable, but conversely, there is no way to mark it as permanently unfocusable. The only way to make it unfocusable would be to not have a label, value, hint, or match any of the `FOCUSABLE_FLAGS`.

## Where it will be useful
In Jetpack Compose, the Material Date Picker sets the <month year> portion as a unfocusable live region whereas the parent is a focusable composable which wraps the <month year> + "Switch to selecting a year" texts. This is so that when the next/previous month buttons are clicked, only then will just the <month year> will be spoken.
<img width="257" alt="Image" src="https://github.com/user-attachments/assets/d09fb279-2b42-4273-8b2f-2424bc008314" />
The resulting semantic tree structure looks like this:
```
(159133)971.View:(99, 816 - 442, 942):GRANULARITY{0}(action:FOCUS/CLEAR_A11Y_FOCUS/CLICK/SCROLL_TO_POSITION):focusable:screenReaderfocusable:clickable:accessibilityFocused
  (159195)971.TextView:(131, 852 - 326, 905):TEXT{March 2025}:CONTENT{March 2025}:GRANULARITY{11}(action:A11Y_FOCUS/SCROLL_TO_POSITION):supportsTextLocation:politeLiveRegion
  (159257)971.View:(347, 847 - 410, 910):CONTENT{Switch to selecting a year}:GRANULARITY{11}(action:A11Y_FOCUS/SCROLL_TO_POSITION)
```
Jetpack Example: https://github.com/ash2moon/jetpack-calendar

In Flutter, the same component cannot be represented the same way Jetpack Compose does without the need for Semantic announcement.
Flutter Example: https://github.com/flutter/flutter/blob/master/examples/api/lib/material/date_picker/show_date_picker.0.dart

## Expected behavior:
There should be a new property named `unfocusable` which forces no other condition to allow for focus.

@reidbaker @jmagman

**Comment by chunhtai** [comment]
I think adding the a accessibilityFocusable (to differentiate SemanticsProperties.focusable which is for keyboard focus) sounds good to me.

We need to make sure we have the equivalent for iOS part though, we should not add single platform only api if possible.

cc @loic-sharma , does iOS has the equivalent for this? ie. makes a single node skip voiceover focus, but everything else like liveregion still works. The only way I am aware of is isAccessibilityElement, but that will likely disable everything.

**Comment by chunhtai** [comment]
Thinking about it more, this property may need to be tristate.
true, false, not provided.

true, false should work as expected.

When not provided, we will default to our current behavior. ie. if has label/value/hint and any non-system semantics semantic actions.

We may also need to wait for SemanticsFlag refactor from @hannah-hyj since the bit map is going to overflow if we added a tristate semantics flag like accessibilityFocusable

**Comment by jmagman** [comment]
I checked with @hannah-hyj, this is waiting on the semantics flag refactoring to land https://github.com/flutter/flutter/issues/166101

**Comment by hannah-hyj** [comment]
i started working on a draft PR for this https://github.com/flutter/flutter/pull/175551
